presto-common/src/main/java/com/facebook/presto/common/type/TimestampType.java

```diff
@@ -82,6 +82,32 @@ public int hashCode()
         return Objects.hash(getClass(), precision);
     }
 
+    /**
+     * Gets the timestamp's number of total seconds.
+     * The epoch second count is a simple incrementing count of seconds where second 0 is 1970-01-01T00:00:00Z.
+     *
+     * Returns:
+     * the total seconds from the epoch of 1970-01-01T00:00:00Z in timestamp
+     */
+    public long getEpochSecond(long timestamp)
+    {
+        return this.precision == MILLISECONDS ?
+                MILLISECONDS.toSeconds(timestamp) :
+                MICROSECONDS.toSeconds(timestamp);
```

Comment:
How about `this.precision.toSeconds(timestamp)`?
If that's the case, I start to wonder do we need this function. sorry for the hassle.

Reply:
> How about `this.precision.toSeconds(timestamp)`?

Oh, it's better, I have fixed it. Thanks a lot for your suggestion that lead to the continuous improvement in our implementation.

I think this function still has some value to provide the utility logic and corresponding comment, for the usage `TimestampType.getPrecision().toSeconds(long)` outside is not so straightforward. What do you think?

Reply:
sure! sold.